src/accommodation/accommodation.service.ts

```diff
@@ -169,11 +188,44 @@ export class AccommodationService {
     return accommodation;
   }
 
-  async addFileToAccommodation(id: string, file: any, ownerId: string): Promise<any> {
+  generateRandomImageFileName(mimetype: string) {
+    const randomString = Math.random().toString(36).substring(2, 15);
+    const timestamp = new Date().getTime();
+    return `${timestamp}_${randomString}.${mimetype.split('/')[1]}`;
+  }
+
+  async uploadImageToS3(file: Express.Multer.File): Promise<UploadImageResponse> {
+    const requestBody: UploadImageType = {
+      filename: this.generateRandomImageFileName(file.mimetype),
+      base64Image: file.buffer.toString('base64url'),
+    };
+
+    const { data } = await firstValueFrom(
+      this.httpService
+        .post(process.env.FILE_UPLOAD_URL!, requestBody, {
+          timeout: 20000,
+        })
+        .pipe(
+          catchError((error: AxiosError) => {
+            throw new GlobalException(
+              ErrorsTypes.ACCOMMODATION_FAILED_TO_STORE_IMAGES_TO_S3,
```

Comment:
not the best error type name. The upload file url can be not only S3 but any storage. so change it to more generic

Reply:
What about FAILED_TO_STORE_IMAGES_TO_STORAGE?